File: java2csharp/java/Bixolon/UPOS Compliant/BXLTest/ListDialogFragment.java
package com.bxl.postest;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentManager;

public class ListDialogFragment extends DialogFragment {
	
	private static final String KEY_TITLE = "title";
	private static final String KEY_ITEMS = "items";
	private static final String TAG = ListDialogFragment.class.getSimpleName();
	
	interface OnClickListener {
		void onClick(String title, String text);
	}
	
	private OnClickListener onClickListener;
	
	static void showDialog(FragmentManager manager, String title, String[] items) {
		ListDialogFragment listDialogFragment = new ListDialogFragment();
		Bundle args = new Bundle();
		args.putString(KEY_TITLE, title);
		args.putStringArray(KEY_ITEMS, items);
		listDialogFragment.setArguments(args);
		listDialogFragment.show(manager, TAG);
	}
	
	@Override
	public void onAttach(Activity activity) {
		super.onAttach(activity);
		
		try {
			onClickListener = (OnClickListener) activity;
		} catch (ClassCastException e) {
			e.printStackTrace();
			throw new ClassCastException(activity.toString()
					+ " must implement ListDialogFragment.OnClickListener");
		}
	}

	@Override
	@NonNull
	public Dialog onCreateDialog(Bundle savedInstanceState) {
		Bundle args = getArguments();
		final String title = args.getString(KEY_TITLE);
		final String[] items = args.getStringArray(KEY_ITEMS);
		
		AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
		builder.setTitle(title).setItems(items, new DialogInterface.OnClickListener() {

			@Override
			public void onClick(DialogInterface dialog, int which) {
				onClickListener.onClick(title, items[which]);
			}
		});
		return builder.create();
	}
}

File: java2csharp/csharp/Bixolon/UPOS Compliant/BXLTest/ListDialogFragment.cs
namespace com.bxl.postest
{

	using Activity = android.app.Activity;
	using AlertDialog = android.app.AlertDialog;
	using Dialog = android.app.Dialog;
	using DialogInterface = android.content.DialogInterface;
	using Bundle = android.os.Bundle;
	using NonNull = android.support.annotation.NonNull;
	using DialogFragment = android.support.v4.app.DialogFragment;
	using FragmentManager = android.support.v4.app.FragmentManager;

	public class ListDialogFragment : DialogFragment
	{

		private const string KEY_TITLE = "title";
		private const string KEY_ITEMS = "items";
		private static readonly string TAG = typeof(ListDialogFragment).Name;

		internal interface OnClickListener
		{
			void onClick(string title, string text);
		}

		private OnClickListener onClickListener;

		internal static void showDialog(FragmentManager manager, string title, string[] items)
		{
			ListDialogFragment listDialogFragment = new ListDialogFragment();
			Bundle args = new Bundle();
			args.putString(KEY_TITLE, title);
			args.putStringArray(KEY_ITEMS, items);
			listDialogFragment.Arguments = args;
			listDialogFragment.show(manager, TAG);
		}

		public override void onAttach(Activity activity)
		{
			base.onAttach(activity);

			try
			{
				onClickListener = (OnClickListener) activity;
			}
			catch (System.InvalidCastException e)
			{
				Console.WriteLine(e.ToString());
				Console.Write(e.StackTrace);
				throw new System.InvalidCastException(activity.ToString() + " must implement ListDialogFragment.OnClickListener");
			}
		}

//JAVA TO C# CONVERTER TODO TASK: Most Java annotations will not have direct .NET equivalent attributes:
//ORIGINAL LINE: @Override @NonNull public android.app.Dialog onCreateDialog(android.os.Bundle savedInstanceState)
		public override Dialog onCreateDialog(Bundle savedInstanceState)
		{
			Bundle args = Arguments;
//JAVA TO C# CONVERTER WARNING: The original Java variable was marked 'final':
//ORIGINAL LINE: final String title = args.getString(KEY_TITLE);
			string title = args.getString(KEY_TITLE);
//JAVA TO C# CONVERTER WARNING: The original Java variable was marked 'final':
//ORIGINAL LINE: final String[] items = args.getStringArray(KEY_ITEMS);
			string[] items = args.getStringArray(KEY_ITEMS);

			AlertDialog.Builder builder = new AlertDialog.Builder(Activity);
			builder.setTitle(title).setItems(items, new OnClickListenerAnonymousInnerClassHelper(this, title, items));
			return builder.create();
		}

		private class OnClickListenerAnonymousInnerClassHelper : DialogInterface.OnClickListener
		{
			private readonly ListDialogFragment outerInstance;

			private string title;
			private string[] items;

			public OnClickListenerAnonymousInnerClassHelper(ListDialogFragment outerInstance, string title, string[] items)
			{
				this.outerInstance = outerInstance;
				this.title = title;
				this.items = items;
			}


			public override void onClick(DialogInterface dialog, int which)
			{
				outerInstance.onClickListener.onClick(title, items[which]);
			}
		}
	}

}